docs/features/nullable-reference-types.md

```diff
@@ -22,69 +22,8 @@ Dictionary<string, object?>? OptDictionaryOptValues; // dictionary may be null,
 ```
 A warning is reported when annotating a reference type with `?` outside a `#nullable` context.
 
-In metadata, nullable reference types are annotated with a `[Nullable]` attribute.
-```c#
-namespace System.Runtime.CompilerServices
-{
-    [AttributeUsage(
-        AttributeTargets.Class |
-        AttributeTargets.GenericParameter |
-        AttributeTargets.Event | AttributeTargets.Field | AttributeTargets.Property |
-        AttributeTargets.Parameter | AttributeTargets.ReturnValue,
-        AllowMultiple = false)]
-    public sealed class NullableAttribute : Attribute
-    {
-        public readonly byte[] NullableFlags;
-    
-        public NullableAttribute(byte b) 
-        {
-            NullableFlags = new byte[] { b };
-        }
-        
-        public NullableAttribute(byte[] b)
-        {
-            NullableFlags = b;
-        }
-    }
-}
-```
-
-Each type reference is accompanied by a NullableAttribute with an array of bytes, where 0 is Oblivious, 1 is NotAnnotated and 2 is Annotated.
-All value types are marked with flag 0 (oblivious).
-
-To optimize trivial cases the attribute can be omitted, or instead can be replaced with an attribute that takes a single byte value rather than an array.  
+[Metadata representation](nullable-metadata.md)
 
-Trivial/optimized cases:
-1)	All parts are NotAnnotated – a NullableAttribute with a single value 1 (rather than an array of 1s)
-2)	All parts are Annotated - a NullableAttribute with a single value 2 (rather than an array of 2s)
-3)	All parts are Oblivious – the attribute is omitted, this matches how we interpret the lack of an attribute in legacy assemblies.
-    For completeness, we would also recognize a NullableAttribute with a single value 0 (rather than an array of 0s),
-    but compiler will never emit an attribute like this. 
-
-NullableAttribute(1) should be placed on a type parameter definition that has a `class!` constraint.
-NullableAttribute(2) should be placed on a type parameter definition that has a `class?` constraint.
-NullableAttribute(2) should be placed on a type parameter definition that has no type constraints, `class`, `struct` and `unmanaged` constraints and
-is declared in a context where nullable type annotations are allowed, that is eqivalent to having an `object?` constraint.
-Other forms of NullableAttribute are not emitted on type parameter definitions and are not specially recognized on them.
-
-The `NullableAttribute` type declaration is synthesized by the compiler if it is not included in the compilation, but is needed to produce the output.
-
-```c#
-// C# representation of metadata
-[Nullable(2)]
```

Comment:
We might want to preserve the examples, perhaps move them to the other file.

Reply:
Added some examples.

---
In reply to: [290459329](https://github.com/dotnet/roslyn/pull/36152#discussion_r290459329) [](ancestors = 290459329)